k4e-agent-sti.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+
+set -e
+
+# The scripts performs the following steps on rhel-8.5:
+# * Install imagebuilder 
+# * Create rpms for yggdrasil and k4e-device-worker
+# * Publish rpms as yum repo to be consumed by imagebuilder
+# * Publish aarch64 rhel-image for edge-device
+
+# The script assumes the machine is registered via subscription-manager
+if ! subscription-manager refresh; then
+   echo  "The machine is not registered."
+   exit 1
+fi
+
+#---------------------------
+# Install osbuild components
+#---------------------------
+dnf install -y osbuild-composer composer-cli cockpit-composer bash-completion
+systemctl enable osbuild-composer.socket --now
+systemctl enable cockpit.socket --now
+systemctl start firewalld
+firewall-cmd --add-service=cockpit && firewall-cmd --add-service=cockpit --permanent
+source  /etc/bash_completion.d/composer-cli
+
+#-----------------------------------
+# Build packages for k4e from source
+#-----------------------------------
+mkdir /home/builder
+
+dnf install -y dbus-devel systemd-devel git golang rpm-build
+rm -rf ~/rpmbuild/rpmbuild/*
+
+# Build yggdrasil rpm
+git clone https://github.com/jakub-dzon/yggdrasil.git /home/builder/yggdrasil
+cd /home/builder/yggdrasil
+export CGO_ENABLED=0
+
+export ARCH=$(uname -i)
+GOPROXY=proxy.golang.org,direct PWD=$PWD spec=$PWD outdir=$PWD make -f .copr/Makefile srpm
+rpm -ihv $(ls -ltr yggdrasil-*.src.rpm | tail -n 1 | awk '{print $NF}')
+if [ "$ARCH" = "aarc64" ]; then
+    # Turn ELF binary stripping off in %post
+    sed -i '1s/^/%global __os_install_post %{nil}/' ~/rpmbuild/SPECS/yggdrasil.spec
+fi
+rpmbuild -bb ~/rpmbuild/SPECS/yggdrasil.spec --target $ARCH
+
+# Build k4e-device-worker rpm
+git clone https://github.com/jakub-dzon/k4e-device-worker.git /home/builder/k4e-device-worker
+cd /home/builder/k4e-device-worker
+if [ "$ARCH" = "aarc64" ]; then
+    make build-arm64
+    make rpm-arm64
+else
+    make build
+    make rpm
+fi
+
+#---------------------------
+# Create packages repository
+#---------------------------
+dnf install -y createrepo httpd
+systemctl enable --now httpd
+firewall-cmd --zone=public --add-port=80/tcp --permanent
+firewall-cmd --reload
+
+mkdir /var/www/html/k4e-repo
+cp /root/rpmbuild/RPMS/${ARCH}/*.${ARCH}.rpm /var/www/html/k4e-repo/
+createrepo /var/www/html/k4e-repo/
```

Comment:
We create the repo here do you think it makes sense to make the compose aware by running:
```
composer-cli sources add repo.toml
```

with 
```
cat << EOF > repo.toml
id = "k4e"
name = "k4e"
type = "yum-baseurl"
url = "http://127.0.0.1/k4e-repo"
check_gpg = false
check_ssl = false
system = false
EOF
```

Reply:
It is already part of the r4e-image.sh script https://github.com/ydayagi/r4e/pull/6/files#diff-67c03e8e4fd3dbc92474714d7be3166f479d2c1efb7827b8e04a4c3916a4b05dR54